src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -118,25 +128,16 @@ private static bool TryGetSpecialTypeAssigningCode(ITypeSymbol type, WellKnownTy
 
     private static bool TryGetExplicitFromJsonBody(IParameterSymbol parameter,
         WellKnownTypes wellKnownTypes,
-        [NotNullWhen(true)] out string? assigningCode,
         out bool isOptional)
     {
-        assigningCode = null;
         isOptional = false;
-        if (parameter.HasAttributeImplementingInterface(wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromBodyMetadata), out var fromBodyAttribute))
-        {
-            foreach (var namedArgument in fromBodyAttribute.NamedArguments)
-            {
-                if (namedArgument.Key == "AllowEmpty")
-                {
-                    isOptional |= namedArgument.Value.Value is true;
-                }
-            }
-            isOptional |= (parameter.NullableAnnotation == NullableAnnotation.Annotated || parameter.HasExplicitDefaultValue);
-            assigningCode = $"await GeneratedRouteBuilderExtensionsCore.TryResolveBody<{parameter.Type}>(httpContext, {(isOptional ? "true" : "false")})";
-            return true;
+        if (!parameter.HasAttributeImplementingInterface(wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromBodyMetadata), out var fromBodyAttribute))
+        {
+            return false;
         }
-        return false;
+        isOptional |= fromBodyAttribute.TryGetNamedArgumentValue<bool>("AllowEmpty", out var allowEmptyValue) && allowEmptyValue;
```

Comment:
How does this work with something like the Mvc `AllowEmpty` property?

https://github.com/dotnet/aspnetcore/blob/ce8e0ce086b0be93d25b985e1c4cdf3fe6d7bc21/src/Mvc/Mvc.Core/src/FromBodyAttribute.cs#L26-L30

My understanding is that the attribute doesn't have a ctor parameter, and instead has this _other_ parameter `EmptyBodyBehavior` that affects the `AllowEmpty` property (which implements the interface explicitly).

It would seem that in order for the Mvc FromBody attribute to work, you need an actual instance of the attribute to call the `AllowEmpty` property on.

Reply:
https://github.com/dotnet/aspnetcore/issues/46639

I think this should actually be easy to do in this PR...